### DISCUSSION TITLE
Alert suppression from wazuh

### DISCUSSION THREAD

**Discussion by GamblerAN** [Security-Onion-Solutions/securityonion#7258] [Unsupported Versions] [answered]
HI!
I am trying to suppress alerts generated from wazuh logs. 
So, i have alert PAM: Login session closed
According to manual, to suppress some alerts i need to make changes in /opt/so/rules/hids/local_rules.xml
So i have created rule: 
` <rule id="100009" level="0" noalert="1">`
   ` <if_sid>5501</if_sid>`
   ` <regex>sudo: pam_unix\(sudo:session\): session closed for user root</regex>`
    `<description>FP</description>`
 ` </rule>`

so-wazuh-restart

that does not solve the problem. 

In addition i have tried several methods. changing the line like that:

`<match>sudo: pam_unix(sudo:session): session closed for user root</match>`
`<match>sudo: pam_unix\(sudo:session\): session closed for user root</match>`
`<match>session closed for user root</match>`

The alert still appear.  
What i am doing wrong?

SO version 2.3.91

**Comment by DebianGuru** [comment]
I think you need to add 'overwrite="yes" noalert="1"' to your stanza in your local_rules.xml file.  I usually just find the original rule, copy it, then paste it into /opt/so/rules/hids/local_rules.  For instance, I don't want an alert every time log files rotate so I do:

` grep -R rotated  /opt/so/rules/hids/ruleset/rules`


which returns.

```
/opt/so/rules/hids/ruleset/rules/0015-ossec_rules.xml:    <match>^ossec: File rotated </match>
/opt/so/rules/hids/ruleset/rules/0015-ossec_rules.xml:    <description>Log file rotated.</description>
```
That tells me the rules is in the 0015-ossec_rules.xml.


I type: 

`nano /opt/so/rules/hids/ruleset/rules/0015-ossec_rules.xml`

I then copy  the stanza: (The formatting gets changed when I put it into this web page, but you'll get the idea)

`
 <rule id="591" level="3">
    <if_sid>500</if_sid>
    <match>^ossec: File rotated </match>
    <description>Log file rotated.</description>
  <group>pci_dss_10.5.2,pci_dss_10.5.5,gpg13_10.1,gdpr_II_5.1.f,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.9,tsc_CC6.1,tsc_CC7.2,tsc_CC7.3,tsc_PI1.4,ts>
</rule>`

I then paste that into the local_rules.xml file as  and type 
`nano /opt/so/rules/hids/local_rules.xml` to add 'overwrite="yes" noalert="1"' to the first line:


```
<rule id="591" level="3" overwrite="yes" noalert="1">
    <if_sid>500</if_sid>
    <match>^ossec: File rotated </match>
    <description>Log file rotated.</description>
    
<group>pci_dss_10.5.2,pci_dss_10.5.5,gpg13_10.1,gdpr_II_5.1.f,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.9,tsc_CC6.1,tsc_CC7.2,tsc_CC7.3,tsc_PI1.4,ts>
>   </rule>
```


Finally, restart wazuh. 

`so-wazuh-restart`

I hope that helps.  If not, I'm afraid I won't be much more help.  I'm no expert, but I have used this several times.

**Comment by GamblerAN** [comment]
Hi!
Thanks for answer. 
Your example totally removes any alerts of Log file rotation. 
My goal is to suppress only some certain alerts that match the criteria. 
In my case, i want to continue to receive "PAM: Login session closed" alerts, but suppress them if the message will contain "sudo: pam_unix(sudo:session): session closed for user root"  And may be IP or hostname in addition.

**Reply by chergett** [reply]
Look here for the options and syntax for wazuh rules:  https://documentation.wazuh.com/3.13/user-manual/ruleset/ruleset-xml-syntax/rules.html

**Reply by GamblerAN** [reply]
So, you think that my problem is related to syntax? But i have tried several variants using `<regex>` or `<match>` as i showed above. 
And most simplest one `<match>session closed for user root</match>`
The source message looks like this: ` Feb 22 22:43:04 mail sudo: pam_unix(sudo:session): session closed for user root`
There is the mistake?

**Comment by wildlyunder** [comment]
Note: Suppression needs to be done on each device (Master or FWD node) which receives logs.

There is currently no solution to do this centrally from the master server.

**Reply by GamblerAN** [reply]
Hi,
thanks for your answer, but i have single server installation.

**Comment by dougburks** [comment]
If your goal is to modify when Wazuh generates the `PAM: Login session closed` alert, then start by looking at that rule itself:

```
   <rule id="5502" level="3">
     <if_sid>5500</if_sid>
     <match>session closed for user </match>
     <description>PAM: Login session closed.</description>
     <group>pci_dss_10.2.5,gpg13_7.8,gpg13_7.9,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
```

Notice that the `rule id` is `5502`. Therefore, if you want to create a child rule that modifies 5502, you would want to use `<if_sid>5502</if_sid>` rather than `<if_sid>5501</if_sid>` as shown in your original post.

**Reply by GamblerAN** [reply]
Thanks, that's correct. I have also rule for `PAM: Login session opened` which has rule id `5501`. I have copied it for `PAM: Login session closed` and forgot to change the id.
I have changed ID to `5502`, but this does not helped. After `so-wazuh-restart`, Alerts still appear with original `level 3`. 
Tried `<match>` and `<regex>`.  
Regex expression doublecheck on https://regex101.com/, to be sure that it should work.

**Comment by GamblerAN** [comment]
I should also mention that non of rules created `local_rules.xml` are working. The rule mentioned above was just an example. May be there is some `enable` switch for that? Somewhere in the main config?

**Comment by dougburks** [comment, marked as answer]
Let's start by using `ossec-logtest` with a default `local_rules.xml`:
```
[doug@securityonion ~]$ sudo docker exec -it so-wazuh /var/ossec/bin/ossec-logtest
2022/02/24 17:52:49 ossec-testrule: INFO: Started (pid: 2298).
ossec-testrule: Type one log per line.

Feb 24 17:46:19 securityonion sshd[37140]: pam_unix(sshd:session): session closed for user doug


**Phase 1: Completed pre-decoding.
       full event: 'Feb 24 17:46:19 securityonion sshd[37140]: pam_unix(sshd:session): session closed for user doug'
       timestamp: 'Feb 24 17:46:19'
       hostname: 'securityonion'
       program_name: 'sshd'
       log: 'pam_unix(sshd:session): session closed for user doug'

**Phase 2: Completed decoding.
       decoder: 'pam'
       dstuser: 'doug'

**Phase 3: Completed filtering (rules).
       Rule id: '5502'
       Level: '3'
       Description: 'PAM: Login session closed.'
**Alert to be generated.
```

Now let's add the following rule to `local_rules.xml`:
```
  <rule id="100002" level="1">
    <if_sid>5502</if_sid>
    <match>doug</match>
    <description>ignore logins from doug</description>
  </rule>
```

Finally, let's re-run ossec-logtest:
```
[doug@securityonion ~]$ sudo docker exec -it so-wazuh /var/ossec/bin/ossec-logtest
2022/02/24 17:54:26 ossec-testrule: INFO: Started (pid: 2305).
ossec-testrule: Type one log per line.

Feb 24 17:46:19 securityonion sshd[37140]: pam_unix(sshd:session): session closed for user doug


**Phase 1: Completed pre-decoding.
       full event: 'Feb 24 17:46:19 securityonion sshd[37140]: pam_unix(sshd:session): session closed for user doug'
       timestamp: 'Feb 24 17:46:19'
       hostname: 'securityonion'
       program_name: 'sshd'
       log: 'pam_unix(sshd:session): session closed for user doug'

**Phase 2: Completed decoding.
       decoder: 'pam'
       dstuser: 'doug'

**Phase 3: Completed filtering (rules).
       Rule id: '100002'
       Level: '1'
       Description: 'ignore logins from doug'
**Alert to be generated.
```

**Reply by dougburks** [reply]
I've added this to https://docs.securityonion.net/en/2.3/wazuh.html#child-rules.
